From 0c57eaaf1981664e8de87be64b8d19af9e00fef5 Mon Sep 17 00:00:00 2001
From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Fri, 3 Dec 2010 22:16:58 +0100
Subject: ASoC: ABE DSP: Update mixer switch to enable/disable ABE gains

Mixer switch controls are not associated to any gain controls. ABE
enable/disable functions should be called when the switch is modified.

Change-Id: Icbd3da69326b96bedd48654213179bb307f2826d
Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
Signed-off-by: Chris Kelly <c-kelly@ti.com>
---
 sound/soc/omap/omap-abe-dsp.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

Index: Kernel/sound/soc/omap/omap-abe-dsp.c
===================================================================
--- Kernel.orig/sound/soc/omap/omap-abe-dsp.c	2011-01-19 00:14:44.212071001 +0530
+++ Kernel/sound/soc/omap/omap-abe-dsp.c	2011-01-19 00:14:48.743323287 +0530
@@ -398,9 +398,11 @@
 	if (ucontrol->value.integer.value[0]) {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 1);
+		abe_enable_gain(MIXDL2, mc->reg);
 	} else {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 0);
+		abe_disable_gain(MIXDL2, mc->reg);
 	}
 
 	pm_runtime_put_sync(&pdev->dev);
@@ -422,9 +424,11 @@
 	if (ucontrol->value.integer.value[0]) {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 1);
+		abe_enable_gain(MIXDL1, mc->reg);
 	} else {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 0);
+		abe_disable_gain(MIXDL1, mc->reg);
 	}
 	pm_runtime_put_sync(&pdev->dev);
 
@@ -446,9 +450,11 @@
 	if (ucontrol->value.integer.value[0]) {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 1);
+		abe_enable_gain(MIXVXREC, mc->reg);
 	} else {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 0);
+		abe_disable_gain(MIXVXREC, mc->reg);
 	}
 	pm_runtime_put_sync(&pdev->dev);
 
@@ -470,9 +476,11 @@
 	if (ucontrol->value.integer.value[0]) {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 1);
+		abe_enable_gain(MIXAUDUL, mc->reg);
 	} else {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 0);
+		abe_disable_gain(MIXAUDUL, mc->reg);
 	}
 	pm_runtime_put_sync(&pdev->dev);
 
@@ -568,9 +576,11 @@
 	if (ucontrol->value.integer.value[0]) {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 1);
+		abe_enable_gain(MIXSDT, mc->reg);
 	} else {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 0);
+		abe_disable_gain(MIXSDT, mc->reg);
 	}
 	pm_runtime_put_sync(&pdev->dev);
 
